### PR TITLE
Add jupyter_app_launcher

### PR DESCRIPTION
**Pull request by trungleduc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by trungleduc** [comment]
@conda-forge-admin, please ping team

File: recipes/jupyter_app_launcher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8d31694b6921e29f5f587f89764a0f1a7c61259cb01f14c62356def9cd31d787
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (ocefpaf) [inline recipes/jupyter_app_launcher/meta.yaml:14]:
```suggestion
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
```

File: recipes/jupyter_app_launcher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+requirements:
+  host:
+    - python
```
Comment (ocefpaf) [inline recipes/jupyter_app_launcher/meta.yaml:18]:
```suggestion
    - python >=3.7
```

File: recipes/jupyter_app_launcher/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - pip
+    - jupyter-packaging >=0.12
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/jupyter_app_launcher/meta.yaml:22]:
```suggestion
    - python >=3.7
```

File: recipes/jupyter_app_launcher/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    - voila >=0.3.0,<0.4
```
Comment (ocefpaf) [inline recipes/jupyter_app_launcher/meta.yaml:23]: The metadata upstream does not constrain it to <0.4
```suggestion
    - voila >=0.3.0
```

Comment (trungleduc) [reply recipes/jupyter_app_launcher/meta.yaml:23]: I'm playing safe here since we are going do a major breaking change in `Voila` 0.4

**Comment by trungleduc** [comment]
Thanks @ocefpaf for reviewing this PR! Applied changes in the last commit.

**Pull request merged**